src/messages/test_proposals.rs

```diff
@@ -120,10 +119,39 @@ fn proposal_queue_functions() {
         assert!(!proposal_queue.contains(invalid_proposal_id_list));
 
         // Get filtered proposals
-        let filtered_proposals =
-            proposal_queue.filtered_queued_proposals(valid_proposal_id_list, ProposalType::Add);
+        let filtered_proposals = proposal_queue.filtered_by_type(ProposalType::Add);
         for filtered_proposal in filtered_proposals {
             assert!(filtered_proposal.proposal().is_type(ProposalType::Add));
         }
     }
 }
+
+/// This test encodes and decodes the `ProposalOrRef` struct and makes sure the
+/// decoded values are the same as the original
+#[test]
+fn proposals_codec() {
```

Comment:
We should add a fuzzing target for this as well.

Reply:
Added.